src/components/stepforms/FromDateStepForm.vue

```diff
@@ -59,5 +106,13 @@ export default class FromDateStepForm extends BaseStepForm<FromDateStep> {
     }
     this.editedStep.column = colname;
   }
+
+  updateStepFormat(newFormat: FormatOption) {
+    if (newFormat.format === 'custom') {
+      this.editedStep.format = '';
+    } else {
+      this.editedStep.format = newFormat.format;
+    }
+  }
```

Comment:
@rhuille I've tried to implement your suggestion which I agree looked more elegant, but actually it's not that easy because `editedStep` comes from the base class `BaseStepForm` so we cannot define it as a computed in this inherited class.

Also, about what you spotted as a complex computed, I do need to check the value of `editedStep` format in all cases so that my data is set correctly when I need to edit a step for example. So be it a data or a computed, I will not be able to get rid of the flow control implemented in the computed, which is not that complicated in my opinion though :)

Are you OK to let the code as it is or do you want to suggest something also?